### PR TITLE
Add tags to aws ecs task

### PR DESCRIPTION
**Pull request by ezmac**
##### SUMMARY

Adds support for tags to ecs_task.  Partially addresses #51122 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ecs_task

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ecs_task.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ecs_task.py]: Can this just be tags for consistency with every other AWS module?

File: lib/ansible/modules/cloud/amazon/ecs_task.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ecs_task.py]: I get that long format task ARNS might be a prerequisite for tags but might be better say e.g.

> "Cannot set task tags: long format task arns are required to set tags"

File: lib/ansible/modules/cloud/amazon/ecs_task.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ecs_task.py]: Probably nicer to use 

```
msg=missing_required_lib("botocore >= 1.12.46", reason="to use tags")
```

File: lib/ansible/modules/cloud/amazon/ecs_task.py
Comment (ezmac) [reply lib/ansible/modules/cloud/amazon/ecs_task.py]: It should have been.  My syntax highlighter got confused and I did as well.

Thanks for your time reviewing and the suggestions you gave.

**Comment by ezmac** [comment]
I also moved the check for account support to the same block as the check for boto support so that there's only one place an error is defined.  if you'd rather have it the other way, please let me know.

**Review by willthames — APPROVED** [review]

**Comment by willthames** [comment]
Looks good now @ezmac - I'll run this through the test suite in the next couple of days and it should be good to merge.

File: test/integration/targets/ecs_cluster/playbooks/roles/ecs_cluster/tasks/main.yml
Comment (willthames) [inline test/integration/targets/ecs_cluster/playbooks/roles/ecs_cluster/tasks/main.yml]: This can be expressed far more healthily (i.e. you can see what has gone wrong because you don't have to use no_log) as:

```
      command: aws ecs put-account-setting --name taskLongArnFormat --value disabled
      environment:
        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
        AWS_SESSION_TOKEN: "{{ security_token }}"
        AWS_DEFAULT_REGION: "{{ aws_region }}"
```

**Review by willthames — CHANGES_REQUESTED** [review]

File: test/integration/targets/ecs_cluster/playbooks/roles/ecs_cluster/tasks/main.yml
Comment (willthames) [reply test/integration/targets/ecs_cluster/playbooks/roles/ecs_cluster/tasks/main.yml]: I'll probably just push up my fix for this shortly, so don't worry too much about it :)

**Comment by willthames** [comment]
I've added a couple of commits that make tests pass

There are some comprehensive changes to the testing policies due to the limit of 10 policies per group - so need to consolidate policies to get back under 10.

**Review by willthames — APPROVED** [review]

**Comment by ezmac** [comment]
Well, I may have messed this PR up trying to fix conflicts.  Sorry about that.  I'll see if I can get it cleaned up again.  Until then, feel free to ignore.

**Comment by ezmac** [comment]
there I fixed it?  I'll try to check back in a bit now that the 2.8 release has (hopefully) calmed down.  Let me know if there are any changes needed; I'm not using 2.8 in my daily life yet.

**Comment by willthames** [comment]
Test suite fails for me again, I think I'll need to totally refactor the ECS test suite which will take a few days, I'll probably split out all the test suite changes into a new PR and then this will sit on top.

The conversion from runme.sh to a role using pip/virtualenv is what's needed, similar to the test suite changes in https://github.com/ansible/ansible/pull/57324 - happy to do that work, it'll just take me a few days

**Comment by ezmac** [comment]
No problems here.  It's hard for me to keep up with the pace at which y'all move doing this part time.  Thanks for all the time you've spent cleaning up my work :)

**Comment by willthames** [comment]
I've rebased this on #57716 - this one should be held from merge until that one has got through.

**Review by willthames — CHANGES_REQUESTED** [review]
Hold on merge until #57716 merged

**Comment by ezmac** [comment]
@willthames Would you be willing to pick this back up or should I do the work to get it merge worthy again?

**Comment by willthames** [comment]
@ezmac I've rebased but haven't yet run a test suite on it - I'll try and do that later on

**Review by willthames — APPROVED** [review]

**Review by jillr — APPROVED** [review]
@ezmac looks like an unresolved merge conflict got pushed in hacking/aws_config/testing_policies/security-policy.json, I'm happy to merge once that's cleaned up.

**Comment by ezmac** [comment]
Sorry about that.  Let me know if the way I cleaned it up is problematic.

**Comment by jillr** [comment]
Looks right to me, tests pass locally, thanks @ezmac!

**Pull request merged**
